### PR TITLE
Add reducers and store

### PR DESCRIPTION
**Pull request by Theophileaseh**
Added reducers and created a store

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+let lastId = 0;
+
```
Comment (KDlamini) [inline src/redux/books/books.js:4]: Please define the `ADD_BOOK` and `REMOVE_BOOK` action types. These are the `constants` that will be used by the `action creators` to determine which action needs to be dispatched.

The constants can look like this at the top of your reducer:
image

But ofcos you can name them based on the `actiontype` names you used.

Comment (KDlamini) [inline src/redux/books/books.js:4]: Kindly create and export `Action Creators` in your reducer file. These are  `addBook` and `removeBook` functions that merely return an action object (payload) that will be handled by the reducer.

Your `actiontypes` and `action creators` can look similar to this together:
```
const ADD_BOOK = bookstore/books/ADD_BOOK;
const REMOVE_BOOK = bookstore/books/REMOVE_BOOK;

export const addBook = (payload) => ({
  type: ADD_BOOK,
  payload,
});

export const removeBook = (payload) => ({
  type: REMOVE_BOOK,
  payload,
});
```

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const CHECK_STATUS = 'CHECK_STATUS';
+
```
Comment (KDlamini) [inline src/redux/categories/categories.js:2]: Kindly add an `action creator` for `CHECK_STATUS` similar to what I explained about the `addBook` and `removeBook` functions.

You can do something similar to this:
```
export const checkStatus = (payload) => ({
  type: CHECK_STATUS,
  payload,
});
```

File: src/redux/combineReducers.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { combineReducers } from 'redux';
+
```
Comment (KDlamini) [inline src/redux/combineReducers.js:2]: Kindly install `redux` as well as `npm install redux`. If you look at your `import` for `combineReducers` and `createStore `, they are from `redux` and if `redux` is not installed you will run into errors in the future for a missing package.

NOTE: `redux` and `react-redux` are two separate packages. I know that you've installed `react-redux`, but you'll also need to install `redux` too to get your imports to work.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @Theophileaseh 🙋‍♀️,  

Great work! 💯 
You have done a great job setting up redux using the duck pattern. 🎊 🥳 
There are some issues that you still need to work on improving your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 
✔️  Correct PR title
✔️  Correct use of Gitflow 
✔️  Commit messages are meaningful
✔️  No linter errors

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @Theophileaseh , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**